src/platform/monitoring/DowntimeNotification/components/DowntimeApproaching.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Modal from '@department-of-veterans-affairs/formation/Modal';
+import serviceStatus from '../config/serviceStatus';
+import dismissedDowntimeNotifications from '../util/dismissedDowntimeNotifications';
+import DowntimeNotificationWrapper from './Wrapper';
+
+export default class DowntimeApproaching extends React.Component {
+  constructor(props) {
+    super(props);
+    dismissedDowntimeNotifications.setup();
+    this.state = {
+      modalDismissed: dismissedDowntimeNotifications.contains(props.appTitle),
+      cache: {}
+    };
+  }
+
+  dismissModal = () => {
+    dismissedDowntimeNotifications.push(this.props.appTitle);
+    this.setState({ modalDismissed: true });
+  }
+
+  render() {
+    const {
+      startTime,
+      endTime,
+      appTitle,
+      children,
+      content
+    } = this.props;
+
+    let downtimeNotification = null;
+    if (!this.state.modalDismissed) {
+      downtimeNotification = (
```

Comment:
I think declarative code is easier to read if these partials are used inline in the jsx. This could be accomplished with a ternary that returns `null` on `false`.

Reply:
I sometimes use the ternary operator but unless it's a very short conditional I generally prefer this style.